File: app/src/main/java/com/jack/autostart/utils/AppLauncherUtils.java
package com.jack.autostart.utils;

import android.content.Context;
import android.content.Intent;

import com.jack.autostart.app.BaseApplication;

public class AppLauncherUtils {

    public static void launchAppWithPackageName(String packageName) {
        Context context = BaseApplication.getContext();
        Intent intent = context.getPackageManager().getLaunchIntentForPackage(packageName);
        if (intent != null) {
            context.startActivity(intent);
        }
    }
}

File: app/src/main/java/com/jack/autostart/utils/AppLauncherUtils.kt
package com.jack.autostart.utils

import com.jack.autostart.app.BaseApplication

object AppLauncherUtils {

    fun launchAppWithPackageName(packageName: String) {
        val context = BaseApplication.context
        val intent = context.packageManager.getLaunchIntentForPackage(packageName)
        if (intent != null) {
            context.startActivity(intent)
        }
    }

}